File: src/equiv/ReverseInteger1.ml
module JB = Javalib_pack.JBasics
module QID = QualifiedIdentity
module E = Expr

let _ =
  let x = E.Var (Var.Mk (QID.QID ["x"], Type.Int)) in

  Equiv.run "ReverseInteger1"
            [(JB.TBasic `Int)]
            (JB.TBasic `Int)
            "reverse1"
            "reverse2"
            [x]
            [E.mk_ige x (E.Int 0)]

File: benchmark/equiv/ReverseInteger1.java
public class ReverseInteger1 {
  static public int reverse1(int x) {
    int res = 0;
    while (x > 0) {
      int mod = x % 10;
      x = x / 10;
      res = res * 10 + mod;
    }
    return res;
  }

  static public int reverse2(int x) {
    int rev = 0;
    while(x != 0){
      rev = rev*10 + x%10;
      x = x/10;
    }

    return rev;
  }
}
